test/test-generate-request-details.js

```diff
@@ -243,4 +244,26 @@ suite('Test Generate Request Details', function() {
     assert.equal(details.headers.Topic, extraOptions.headers.Topic);
     assert.equal(details.headers.Urgency, extraOptions.headers.Urgency);
   });
+
+  test('Audience contains port', function() {
+    const subscription = {
+      endpoint: 'http://example.com:4242/life-universe-and-everything'
+    };
+
+    const extraOptions = {
+      vapidDetails: {
+        subject: 'mailto:example@example.com',
+        publicKey: vapidKeys.publicKey,
+        privateKey: vapidKeys.privateKey
```

Comment:
@marco-c These are taken from `vapidKeys` available in the parent closure of `suite` function call, and there they are generated using `generateVAPIDKeys` method of `vapid-helper.js` which already does this.

Since,  I went into their implementation and it's doing both of these things

```js
function generateVAPIDKeys() {
  const curve = crypto.createECDH('prime256v1');
  curve.generateKeys();

  return {
    publicKey: urlBase64.encode(curve.getPublicKey()),
    privateKey: urlBase64.encode(curve.getPrivateKey())
  };
}
```